src/sql-parser/src/parser.rs

```diff
@@ -4519,6 +4526,37 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parse a mutually recursive CTE (`alias ( col1: typ1, col2: typ2, ... ) AS (subquery)`).
+    ///
+    /// The main distinction from `parse_cte` is that the column names and types are mandatory.
+    /// This is not how SQL works for `WITH RECURSIVE`, but we are doing it for now to make the
+    /// query interpretation that much easier.
+    fn parse_mut_rec_cte(&mut self) -> Result<CteMutRec<Raw>, ParserError> {
```

Comment:
I think either this should be `parse_cte_mut_rec` or the type should be `MutRecCte`.

Reply:
with with `parse_cte_mut_rec`!